plugins/sinks/columbus/sink.go

```diff
@@ -33,27 +35,71 @@ func New(c httpClient) plugins.Syncer {
 }
 
 func (s *Sink) Sink(ctx context.Context, configMap map[string]interface{}, in <-chan interface{}) (err error) {
-	if err = utils.BuildConfig(configMap, &s.config); err != nil {
+	var config Config
+	if err = utils.BuildConfig(configMap, &config); err != nil {
 		return plugins.InvalidConfigError{Type: plugins.PluginTypeSink}
 	}
 
 	for data := range in {
-		if err = s.send(data); err != nil {
+		dataMapper := s.getDataMapper(data, config)
```

Comment:
@StewartJingga can we move config to Sink struct, just like we did in metabase extractor?? Since it is being passed as a parameter.

Reply:
@StewartJingga Any thoughts on the above comment? Apart from this LGTM.

Reply:
> @StewartJingga can we move config to Sink struct, just like we did in metabase extractor?? Since it is being passed as a parameter.

oh actually being passed a parameter is not the reason for utilising internal states, unless you find yourself passing it over and over again. But yeah sure both approach should be fine here, will update the PR.

Reply:
updated 🙏 @GrayFlash